lib/dataSource-url.js

```diff
@@ -94,6 +94,9 @@ kettle.dataSource.URL.requestOptions = ["url", "protocol", "host", "hostname", "
 kettle.dataSource.URL.prepareRequestOptions = function (componentOptions, cookieJar, userOptions, permittedOptions, directModel, userStaticOptions, resolveUrl) {
     var staticOptions = fluid.filterKeys(componentOptions, permittedOptions);
     var requestOptions = fluid.extend(true, {headers: {}}, userStaticOptions, staticOptions, userOptions);
+    if (requestOptions.hostname === "localhost" || requestOptions.host === "localhost") {
```

Comment:
Also a tiny test would be helpful - it doesn't need to be particularly realistic, but we should at least have something which verifies that this code path hasn't rotted (and will show up in a coverage check)